registry/storage/driver/hdfsweb/fs.go

```diff
@@ -0,0 +1,554 @@
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	webHdfsVer string = "/webhdfs/v1"
+)
+
+// FileSystem maps fields and functions to HDFS's FileSystem class.
+type FileSystem struct {
+	NameNodeHost string
+	NameNodePort string
+	Username     string
+}
+
+// Create creates a new file and stores its content in HDFS.
+func (fs *FileSystem) Create(
+	data io.Reader,
+	path string,
+	overwrite bool,
+	blocksize uint64,
+	replication uint16,
+	permission os.FileMode,
+	buffersize int) (bool, error) {
+
+	client := &http.Client{}
+	params := map[string]string{
+		"op":          "CREATE",
+		"overwrite":   strconv.FormatBool(overwrite),
+		"blocksize":   strconv.FormatInt(int64(blocksize), 10),
+		"replication": strconv.FormatInt(int64(replication), 10),
+		"permission":  strconv.FormatInt(int64(permission), 8),
+		"buffersize":  strconv.FormatInt(int64(buffersize), 10),
+	}
+
+	u, err := fs.buildRequestURL(path, &params)
+	if err != nil {
+		return false, err
+	}
+
+	// take over default transport to avoid redirect
+	tr := &http.Transport{}
+	req, _ := http.NewRequest("PUT", u.String(), nil)
+	rsp, err := tr.RoundTrip(req)
+	if err != nil {
+		return false, err
+	}
+
+	// extract returned url in header.
+	loc := rsp.Header.Get("Location")
+	u, err = url.ParseRequestURI(loc)
+	if err != nil {
+		return false, fmt.Errorf("FileSystem.Create(%s) - invalid redirect URL from server: %s", u, err.Error())
+	}
+
+	req, _ = http.NewRequest("PUT", u.String(), data)
+	rsp, err = client.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("FileSystem.Create(%s) - bad url: %s", loc, err.Error())
+	}
+
+	if rsp.StatusCode != http.StatusCreated {
+		defer rsp.Body.Close()
+		_, err = responseToHdfsData(rsp)
+		if err != nil {
+			return false, err
+		}
+		return false, fmt.Errorf("FileSystem.Create(%s) - File not created.  Server returned status %v", loc, rsp.StatusCode)
+	}
+
+	return true, nil
+}
+
+//Open opens the specificed Path and returns its content to be accessed locally.
+func (fs *FileSystem) Open(path string, offset, length int64, buffSize int) (io.ReadCloser, error) {
+	client := &http.Client{}
+	params := map[string]string{
+		"op":         "OPEN",
+		"offset":     strconv.FormatInt(offset, 10),
+		"length":     strconv.FormatInt(length, 10),
+		"buffersize": strconv.Itoa(buffSize),
+	}
+
+	u, err := fs.buildRequestURL(path, &params)
+	if err != nil {
+		return nil, err
+	}
+
+	req, _ := http.NewRequest("GET", u.String(), nil)
+	rsp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// possible error
+	if rsp.StatusCode != http.StatusOK {
+		defer rsp.Body.Close()
+		_, err = responseToHdfsData(rsp)
+		if err != nil {
+			return nil, err
+		}
+
+		return nil, fmt.Errorf("Open(%s) - File not opened.  Server returned status %v", path, rsp.StatusCode)
+	}
+
+	return rsp.Body, nil
+}
+
+// Append appends specified data to an existing file.
+func (fs *FileSystem) Append(data io.Reader, path string, buffersize int) (bool, error) {
```

Comment:
This method does exactly what an `io.Writer` would do. It might be better to encapsulate these operations into a single object (io.WriteCloser, perhaps) that can manage this. Doing so will open up a lot of the tools in the `io` package that aren't useful with the way this method is written.